src/libraries/Microsoft.PowerFx.Interpreter/Marshal/LazyMarshalledTypeMetadata.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    [ThreadSafeImmutable]
+    internal class LazyMarshalledTypeMetadata : ILazyTypeMetadata
```

Comment:
This maintains the relatively simple notion of identity for a C# marshalled object. For Relationships, e.g, this would be implemented as the table name and dataset name instead.